models/articles.models.js

```diff
@@ -0,0 +1,30 @@
+const db = require("../db/connection");
+
+exports.selectArticleById = (article_id) => {
+  return db
+    .query("SELECT * FROM articles WHERE articles.article_id = $1;", [
+      article_id,
+    ])
+    .then(({ rows: [article] }) => {
+      if (!article) {
+        return Promise.reject({
+          status: 404,
+          msg: "article does not exist",
+        });
+      }
+      return article;
+    });
+};
+
+exports.selectArticles = () => {
+  return db
+    .query(
+      "SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT OUTER JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;"
+    )
+    .then((result) => {
+      return result.rows.map((article) => {
+        article.comment_count = +article.comment_count;
+        return article;
```

Comment:
Presume this is a way of changing the comment_count data type to number?  I dunno if this is necessary, but can't see any harm!